### PR TITLE
Add `list-restart` icon

### PR DESCRIPTION
**Pull request by danielbayley**
Doubles as `debug-restart-frame` from [codicons](https://microsoft.github.io/vscode-codicons/dist/codicon.html)

### CODE REVIEW

File: icons/list-restart.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M7 12H3" />
+  <path d="M7 18H3" />
+  <path d="M21 6H3" />
+  <path d="M11 12v4h4" />
+  <path d="m11 16 2.5-2.5c1-1 2.3-1.5 3.5-1.5 2.8 0 5 2.2 5 5s-2.2 5-5 5c-1.6 0-3.1-.8-4-2" />
```
Comment (jguddas) [inline icons/list-restart.svg:16]:
```suggestion
   <path d="m11 16 2.5-2.5A5.02 5.02 0 1 1 13 20"/>
```

**Review by ericfennis — CHANGES_REQUESTED** [review]
Can you use the same refresh icon from `database-backup`?

**Comment by karsa-mistmere** [comment]
> Can you use the same refresh icon from `database-backup`?

@ericfennis: There's a related PR, definitely let's keep them consistent. https://github.com/lucide-icons/lucide/pull/1051

**Comment by danielbayley** [comment]
> > Can you use the same refresh icon from `database-backup`?
> 
> @ericfennis: There's a related PR, definitely let's keep them consistent. #1051

I think Illustrator has given me slightly different points (which I’ll fix) but essentially it is the same arrow…?

See also comment: https://github.com/lucide-icons/lucide/pull/1051#discussion_r1169919139

**Comment by ericfennis** [comment]
Currently duplicates changes from #1121.

**Comment by danielbayley** [comment]
> Currently duplicates changes from #1121.

@ericfennis Fixed. Not sure what happened there…

File: icons/list-restart.svg [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M21 6H3" />
+  <path d="M7 12H3" />
+  <path d="M7 18H3" />
+  <path d="M12 18c.9 1.2 2.3 2 4 2 2.8 0 5-2.2 5-5 0-1.2-.4-2.3-1.3-3.2-1.8-1.8-4.6-1.8-6.4 0L11 14" />
```
Comment (karsa-mistmere) [inline icons/list-restart.svg:15]: Same optimization as in #1051
```suggestion
  <path d="M13 20a5 5 0 0 0 9-3 4.5 4.5 0 0 0-4.5-4.5c-1.33 0-2.54.54-3.41 1.41L12 16" />
```

**Pull request merged**